util/suffix.c

```diff
@@ -145,24 +198,30 @@ const char *suffix_dbinary_get(double *value)
 	return "";
 }
 
-uint64_t suffix_binary_parse(const char *value)
+int suffix_binary_parse(const char *str, char **endptr, uint64_t *val)
 {
-	char *suffix;
-	errno = 0;
-	uint64_t ret = strtoull(value, &suffix, 0);
-	if (errno)
+	uint64_t ret;
+	int i;
+
+	ret = strtoull(str, endptr, 0);
+	if (str == *endptr ||
+	    ((ret == ULLONG_MAX) && errno == ERANGE))
+		return -EINVAL;
+
+	if (str == *endptr) {
+		*val = ret;
 		return 0;
+	}
+
+	for (i = 0; i < ARRAY_SIZE(binary_suffixes); i++) {
+		struct binary_suffix *s = &binary_suffixes[i];
 
-	struct binary_suffix *s;
-	for (s = binary_suffixes; s->shift != 0; s++) {
-		if (tolower(suffix[0]) == tolower(s->suffix[0])) {
+		if (tolower(*endptr[0]) == tolower(s->suffix[0])) {
```

Comment:
The character `i` of the suffix "Xi" is not checked but ignored and is it not necessary?

Reply:
Don't know... existing code. But this parser could also get some improvements.

Reply:
I see. Thanks for your replies.

Reply:
I've updated the test so it tests also for the possible second char.